docs/adr/environments-and-ci.md

```diff
@@ -0,0 +1,78 @@
+# ADR 35580f0d-f429-412b-acef-83655e3cab11: Runtime/engine/host/environment support and CI
+
+## Status
+
+Proposed
+
+## Submitters
+
+- @ctcpip
+
+## Decision Owners
+
+- @expressjs/express-tc
+
+## Context
+
+Express and its libraries were specifically designed to run with Node.js (V8). While some of our libraries can run in other environments (e.g. runtimes, engines, browsers), they are not necessarily supported in all environments. Consequently, our CI systems do not include other environments as part of their testing workflows.
+
+Several points were raised during the discussion:
+
+- Cost of running additional CI vs the likelihood of detecting a problem
+- Introducing maintenance overhead and possible coupling to other environments' development lifecycle
+- No JS engine implements ECMAScript 100% correctly; thus, claiming "ES2015 support" does not guarantee correctness across all environments.
+- Environment regressions or language edge cases could break functionality in unpredictable ways that are not practical for us to monitor across all environments.
+
+## Decision
+
+- We will **not** add non-Node.js environment testing to our CI pipelines.
+  - Exceptions may be made iff there is strongly compelling, project-aligned justification.
```

Comment:
```suggestion
  - Exceptions may be made if there is strongly compelling, project-aligned justification.
```

Reply:
https://en.wiktionary.org/wiki/iff   🙂

Reply:
Ok I am confused, is this intentional or an etymology joke 😆 ?

Either way, lets merge the suggestion right?